frontend-maven-plugin/src/main/java/com/github/eirslett/maven/plugins/frontend/mojo/WebpackMojo.java

```diff
@@ -66,6 +85,19 @@ public synchronized void execute(FrontendPluginFactory factory) throws Exception
         boolean incrementalEnabled = incrementalBuildEnabled(buildContext);
         boolean shouldExecute = shouldExecute();
 
+        String nodeVersion = getNodeVersion(workingDirectory, this.nodeVersion, this.nodeVersionFile, project.getArtifactId(), getFrontendMavenPluginVersion());
+
+        if (isNull(nodeVersion)) {
+            throw new LifecycleExecutionException("Node version could not be detected from a file and was not set");
+        }
+
+        if (!NodeVersionHelper.validateVersion(nodeVersion)) {
+            throw new LifecycleExecutionException("Node version (" + nodeVersion + ") is not valid. If you think it actually is, raise an issue");
+        }
+
+        String validNodeVersion = getDownloadableVersion(nodeVersion);
+        factory.loadNodeVersionManager(validNodeVersion);
```

Comment:
How did it get the webpack binary before? Presumably out of `node_modules/.bin`? How did it know the node version, relied on it being in the FMP managed node installation folder before?